### DISCUSSION TITLE
New space for ideas and feature requests in Meteor

### DISCUSSION THREAD

**Discussion by filipenevola** [meteor/meteor#11504] [Announcements]
## We are inaugurating the use of GitHub Discussions in Meteor

This is a space for the community to request Features and to discuss ideas.

Please read [here](https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests) the instructions for new feature suggestions.

### Meteor Software, the company behind Meteor Platform and Meteor Cloud, is going to join **every** discussion and provide clear directions when necessary.

We reinforce that all opinions should be presented here in a polite manner, respecting different opinions and everybody involved.

We are not going to tolerate any type of aggression or misbehavior. Also the discussions need to be productive and looking for the best interest of the community, otherwise the discussion is going to be closed.

##### A note about Meteor Feature Requests repository
[Meteor Feature Requests](https://github.com/meteor/meteor-feature-requests/) repository was created a few years ago when GitHub discussions were not available but now we believe GitHub Discussions in the main repository is the best place for the Meteor Community to request Features and to discuss ideas.

Meteor Feature Requests repository is now archived and in read-only mode. If you have pending Feature Requests there please open a Discussion here under `Ideas` and feel free to link to the old issue in the top and also add relevant information in the description.

We are going to start from a clean slate so we are sure that all the discussions here are still relevant and Meteor Software will provide answers always when necessary.

**Comment by Saeeed-B** [comment]
@filipenevola 
I think the first turning point for the meteor is the expansion of tutorials and examples.
There are many issues that do not have a clear example and do not have a clear explanation.

**Reply by filipenevola** [reply]
Hi @Saeeed-B you could suggest specific tutorials and examples. And also you could help to implement them.

If you need help defining and discussing details you could open a new discussion about a specific one.

It's important to be very specific so we can have a productive discussion.

**Reply by Saeeed-B** [reply]
@filipenevola  There are many topics to set an example.
For example, how to set up on a vps server (other than Galaxy) from zero to one hundred was very difficult for me. Because there are no clear examples.
Setup with the following features:
- On a virtual server (other than Galaxy)
- nginx startup with ssl and proxy
**- Launch mongodb safely and securely and connect it to the software.**
**I still have a problem with this and how many times my database has been hacked and I still have not received a response:**
[https://forums.meteor.com/t/launch-mongodb-with-high-security-on-the-server/56142/4](forums.meteor : launch-mongodb-with-high-security-on-the-server)

And ....

**Reply by Saeeed-B** [reply]
@filipenevola 
Another important issue is the issue of ssr which is really deprived.
There is no comprehensive example that solves the problem in the meteor itself.
Only recently has fast rendering been revived, and it has restored hope.
But ssr must be embedded in the meteor nucleus completely and comprehensively.
more in :
[https://github.com/meteor/meteor-feature-requests/issues/420](https://github.com/meteor/meteor-feature-requests/issues/420)

**Reply by Saeeed-B** [reply]
@filipenevola  I think the ssr issue must have a specific conversation.
Personally, I have so much experience in this issue that it must be discussed in an exclusive interview

**Reply by filipenevola** [reply]
Hi @Saeeed-B I already saw you in a few places talking about SSR, most Meteor apps today are SPAs so they are not relying on SSR that much, only to inject some tags and create some caches.

It would be nice if you could work in some PRs to extend Meteor features to provide more features for devs that really want to rely more heavily on the server.

We are open to new PRs as well, to review, provide feedback, etc. For example, you could start to add a way for data from publications to be available somehow in the server as well.

But as I said in my first reply, we need to have specific suggestions and also people willing to contribute in these areas.

Feel free to open a new discussion about a very specific case so the community can help you to define better your idea for you to work on the PR.

This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

**Comment by technicalbirdVayuz** [comment]
Hi,

Can we use lean Queries?

- https://mongoosejs.com/docs/tutorials/lean.html

For faster queries response?

**Reply by StorytellerCZ** [reply]
Lean queries are specific to mongoose, which Meteor does not uses. Meteor builds on top of the default MongoDB driver.

**Reply by filipenevola** [reply]
You could use any MongoDB lib with Meteor but you should handle the connection and everything, as @StorytellerCZ said in Meteor actual mongo core package Meteor is using plain MongoDB driver and not any lib on top of that.

If you have ideas on how to take advantage of Lean Queries to delivery more value to Meteor users feel free to open a new discussions and explain your plan in details.  

This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

**Comment by mostafiz93** [comment]
I propose following points to give more focus or priority:

1. Opening REST endpoints more easily. 
2. Releasing more stable DDP client so that Android and iOS applications can easily use Meteor methods as API by calling RPC over the DDP client. Currently there is no well maintained DDP for iOS in Swift language. So, iOS application can't use the Meteor backend.

**Reply by filipenevola** [reply]
Hi @mostafiz93, feel free to open a new discussion about your two specific cases so the community can help you to define better your idea for you to work on the PR.

This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

**Comment by satyavh** [comment]


**Reply by matikucharski** [reply]
I think this one is crucial if we want to keep meteor accessible for newcomers. Many of them uses vue as frontend layer.

**Reply by Fraraven** [reply]
I Fully Agree

**Reply by harry-73** [reply]
And more and more are supported vue3 ans vue2 version will have less traction

**Reply by satyavh** [reply]
Moved here: https://github.com/meteor/meteor/discussions/11521

**Reply by filipenevola** [reply]
thanks @satyavh

**Comment by satyavh** [comment]


**Reply by WilliamKelley** [reply]
You've listed some great ideas, but this post is not the correct place to write these requests. You should move each of them to their own Discussion Post 👍🏼

**Reply by satyavh** [reply]
Really? I thought that this is '_the new space for ideas and feature request_' ? There were already multiple proposals and I just added new ones, and each one can be upvoted.

**Reply by WilliamKelley** [reply]
The context to that is _Discussions_ is the "the new space", not this single post. I understand the confusion though, the post could have stated it more explicitly. For confirmation, see part of Filipe's response to all existing replies

> This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

**Reply by WilliamKelley** [reply]
@filipenevola It seems that your original post could benefit from a more explicit message to create new posts for feature requests and hide/delete all current replies once they're migrated.

**Reply by satyavh** [reply]
Ok clear, I will move my requests to the main board today

**Comment by diavrank** [comment]
**Upload files as form-data over DDP**

I have read that exists a way to upload files using **meteor-files** package. However, many of us store files in the server or in a cloud storage but not in a database (where is the place that meteor-files package uses). Currently, if I want to upload files, for example an image, I have to **convert it to base 64 string** and after send this data over a meteor method. Another case is if I want to **upload multiple files** (which comes from an html form), I have to create a **REST endpoint** (I use `maka:rest` :) ) to accept files in form-data.

So, It would be great to have something like this:

```javascript
//Client
Meteor.callWithFormData('methodName',formData,callback);
```

I think that this feature would be revolutionary for **DDP protocol** to compete with **HTTP protocol**, and also this would do development easier.

**Reply by satyavh** [reply]
I'm not sure this is such a good idea, as far as I understand, DDP is a serial protocol, while HTTP is parallel. 
As such I doubt that DDP is actually a better protocol than HTTP for uploading files. Imagine you upload a 1gb file, that would 'block' other DDP requests while it's in progress, while with HTTP you could do parallel chunks and perform other HTTP requests while uploading.
Correct me if I'm wrong!

**Reply by renanccastro** [reply]
Hello @diavrank!
Can you please open a new discussion for this feature request?
We can discuss it better there! Thanks!

**Reply by filipenevola** [reply]
This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

**Comment by filipenevola** [comment]
This discussion in specific is just to introduce GitHub Discussions in Meteor and it's not the right place to ask or discuss specific features. Please read the instructions carefully before creating a new Discussion requesting a feature https://github.com/meteor/meteor/blob/devel/CONTRIBUTING.md#feature-requests

From now on this specific discussion will be locked.
